SpatialGDK/Source/SpatialGDK/Public/Schema/MigrationDiagnostic.h

```diff
@@ -0,0 +1,150 @@
+
+		const VirtualWorkerId LocalVirtualWorkerId = NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId();
+
+		Schema_AddInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_AUTHORITY_WORKER_ID,
+						NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId());
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_LOCKED_ID, NetDriver->LockingPolicy->IsLocked(BlockingActor));
+
+		AActor* NetOwner;
+		VirtualWorkerId NewAuthWorkerId;
+
+		FSpatialLoadBalancingHandler MigrationHandler(NetDriver);
+		FSpatialLoadBalancingHandler::EvaluateActorResult Result =
+			MigrationHandler.EvaluateSingleActor(BlockingActor, NetOwner, NewAuthWorkerId);
+		bool bCanMigrate = Result == FSpatialLoadBalancingHandler::EvaluateActorResult::Migrate;
+
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_EVALUATION_ID,
+					   Result == FSpatialLoadBalancingHandler::EvaluateActorResult::Migrate);
+		Schema_AddInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_DESTINATION_WORKER_ID, NewAuthWorkerId);
+		AddStringToSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_OWNER_ID, GetNameSafe(NetOwner));
+
+		return CommandResponse;
+	}
+
+	// Compare information from the worker that is blocked from migrating a hierarchy with the information from the authoritative
+	// worker over the actor that is blocking the migration and log the results.
+	static FString CreateMigrationDiagnosticLog(USpatialNetDriver* NetDriver, Schema_Object* ResponseObject, AActor* BlockingActor)
+	{
+		check(NetDriver != nullptr);
+		check(NetDriver->Connection != nullptr);
+		check(NetDriver->LockingPolicy != nullptr);
+
+		if (ResponseObject == nullptr)
+		{
+			return FString::Printf(TEXT("Migration diaganostic log failed as response was empty."));
+		}
+
+		VirtualWorkerId AuthoritativeWorkerId = Schema_GetInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_AUTHORITY_WORKER_ID);
+		Worker_EntityId BlockedEntityId = Schema_GetInt64(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_ENTITY_ID);
+		bool bIsReplicated = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_REPLICATES_ID);
+		bool bHasAuthority = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_HAS_AUTHORITY_ID);
+		bool bIsLocked = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_LOCKED_ID);
+		bool bCanMigrate = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_EVALUATION_ID);
+		VirtualWorkerId DestinationWorkerId = Schema_GetInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_DESTINATION_WORKER_ID);
+		FString NetOwnerName = GetStringFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_OWNER_ID);
+
+		FString Reason = FString::Printf(TEXT("Originating worker %i does not have authority of blocking actor. "),
+										 NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId());
+
+		if (bHasAuthority)
```

Comment:
bHasAuthority is expected to be true in that case, so the opposite would be a failure reason.

Reply:
Yes, a good point we need to log that failure case but we also need to log the worker id that has authority as that is important information too.